prow/plugins/branch-updater/branch-updater.go

```diff
@@ -0,0 +1,148 @@
+package branch_updater
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/sirupsen/logrus"
+
+	"k8s.io/test-infra/prow/config"
+	"k8s.io/test-infra/prow/github"
+	"k8s.io/test-infra/prow/pluginhelp"
+	"k8s.io/test-infra/prow/plugins"
+)
+
+const (
+	gitHubContextNameTide        = "tide"
+	gitHubMergeableStateBehind   = "behind"
+	gitHubMergeStateRefreshDelay = 10
+	pluginName                   = "branch-updater"
+	tideContextStatusSuccess     = "success"
+)
+
+var (
+	enabledEvents = []github.PullRequestEventAction{
+		github.PullRequestActionOpened,
+		github.PullRequestActionEdited,
+		github.PullRequestActionReadyForReview,
+		github.PullRequestActionReopened,
+	}
+)
+
+type githubClient interface {
+	UpdatePullRequestBranch(owner, repo string, number int, expectedHeadSha *string) error
+	GetPullRequest(owner, repo string, number int) (*github.PullRequest, error)
+	GetCombinedStatus(org, repo, ref string) (*github.CombinedStatus, error)
+}
+
+func helpProvider(config *plugins.Configuration, enabledRepos []config.OrgRepo) (*pluginhelp.PluginHelp, error) {
+	configInfo := map[string]string{}
+	for _, repo := range enabledRepos {
+		repoUpdateTmpl := "PR branches in repo %s will not be automatically updated."
+		if config.BranchUpdater.ShouldUpdateBranchesForRepo(repo.String()) {
+			repoUpdateTmpl = "PR branches in repo %s will be automatically updated."
+		}
+		configInfo[repo.String()] = fmt.Sprintf(repoUpdateTmpl, repo)
+	}
```

Comment:
i dont really understand this iteration; where does `enabledRepos` come from, and why do you need to check if the enabledRepos are enabled? bad naming?

Reply:
interesting, thanks for sharing!